### PR TITLE
feat: support model selection

### PR DESCRIPTION
**Pull request by stew-ro**

### CODE REVIEW

**Comment by alex-krasn** [comment]
it fails some accessibility tests: 
Screen Shot 2020-07-17 at 12 33 49 PM

**Comment by alex-krasn** [comment]
Models info/columns in modal not scrollable or responsive on x-axis for narrower windows, so not all model info is accessible
Screen Shot 2020-07-17 at 12 37 13 PM
:

File: src/react/components/common/common.scss [hunk 1]
```diff
@@ -182,3 +182,17 @@
   cursor: pointer;
   width: 100%;
 }
+
+.separator-right-pane-main {
+  margin: 0 1rem;
+  div{
```
Comment (alex-krasn) [inline src/react/components/common/common.scss:188]: needs space between the div and bracket -  div {

File: src/react/components/pages/modelCompose/modelCompose.scss [hunk 2]
```diff
@@ -165,7 +166,8 @@ h4 {
 }
 
 .modal-buttons-container {
-    justify-content: space-between;
+    justify-content: flex-end;
+    display: flex;
```
Comment (alex-krasn) [inline src/react/components/pages/modelCompose/modelCompose.scss:170]: I don't think we need 'display: flex' here

File: src/react/components/pages/predict/predictPage.tsx [hunk 3]
```diff
@@ -5,13 +5,16 @@ import React from "react";
 import { connect } from "react-redux";
 import { RouteComponentProps } from "react-router-dom";
 import { bindActionCreators } from "redux";
-import { FontIcon, PrimaryButton, Spinner, SpinnerSize, IconButton, TextField, IDropdownOption, Dropdown} from "@fluentui/react";
+import {
```
Comment (alex-krasn) [inline src/react/components/pages/predict/predictPage.tsx:8]: do we need new line here?

File: src/react/components/pages/predict/predictPage.tsx [hunk 4]
```diff
@@ -5,13 +5,16 @@ import React from "react";
...
+    FontIcon, Selection, PrimaryButton, Spinner, SpinnerSize, IconButton, TextField, IDropdownOption,
+    Dropdown, DefaultButton, Separator, ISelection, SelectionMode
+} from "@fluentui/react";
 import IProjectActions, * as projectActions from "../../../../redux/actions/projectActions";
 import IApplicationActions, * as applicationActions from "../../../../redux/actions/applicationActions";
 import IAppTitleActions, * as appTitleActions from "../../../../redux/actions/appTitleActions";
 import "./predictPage.scss";
 import {
-    IApplicationState, IConnection, IProject, IAppSettings, AppError, ErrorCode,
+    IApplicationState, IConnection, IProject, IAppSettings, AppError, ErrorCode, IRecentModel,
```
Comment (alex-krasn) [inline src/react/components/pages/predict/predictPage.tsx:17]: do we need new line here?

File: src/react/components/pages/predict/predictPage.tsx [hunk 5]
```diff
@@ -131,15 +147,35 @@ export default class PredictPage extends React.Component<IPredictPageProps, IPre
...
+        }
+
+        if (!this.props.project) {
+            const projectId = this.props.match.params["projectId"];
+            if (projectId) {
+                await this.loadProject(projectId);
+            }
+        }
+
+        if (this.props.project?.predictModelId && !(this.props.project?.recentModelRecords) &&
```
Comment (alex-krasn) [inline src/react/components/pages/predict/predictPage.tsx:168]: why do we need round brackets around this.props.project?.recentModelRecords and this.state.couldNotGetRecentModel

File: src/react/components/pages/modelCompose/modelCompose.scss (hunk 2 above)
Comment (stew-ro) [reply src/react/components/pages/modelCompose/modelCompose.scss:170]: yes, we do

File: src/react/components/pages/predict/predictPage.tsx [hunk 6]
```diff
@@ -199,143 +236,188 @@ export default class PredictPage extends React.Component<IPredictPageProps, IPre
...
+                                                            {strings.predict.modelNamePrefix}
+                                                        </span>
+                                                        <span className="model-selection-info-value">
+                                                            {mostRecentModel.modelInfo.modelName}
+                                                        </span>
+                                                    </h6>
+                                                }
+                                            </div>
+                                            <DefaultButton
+                                                style={{width: "80px"}}
```
Comment (alex-krasn) [inline src/react/components/pages/predict/predictPage.tsx:273]: maybe better use className and .scss - it's repeating 3 times

File: src/react/components/pages/modelCompose/modelCompose.scss (hunk 2 above)
Comment (alex-krasn) [reply src/react/components/pages/modelCompose/modelCompose.scss:170]: oh, yes you're right

**Comment by alex-krasn** [comment]
Model list view: now we have unnecessary vertical scrollbar and columns still not accessible (have fixed width and not scrollable/resizable)
Screen Shot 2020-07-17 at 1 20 33 PM

File: src/common/themes.ts [hunk 7]
```diff
@@ -279,6 +304,7 @@ const subMenuPalette = {
 }
 
 
+const rightPaneDefaultButtonTheme = createTheme({palette: rightPaneDefaultButtonPalette});
```
Comment (alex-krasn) [inline src/common/themes.ts:307]: should be ```getRightPaneDefaultButtonTheme```

**Comment by alex-krasn** [comment]
The only other thing -  is that, I think we should put all messages and titles into localization strings. Not blocking issue, though.

File: src/react/components/pages/modelCompose/composeModelView.tsx [hunk 8]
```diff
@@ -3,7 +3,7 @@
 
 import React from "react";
 import { Customizer, IColumn, ICustomizations, Modal, DetailsList, SelectionMode, DetailsListLayoutMode, PrimaryButton, TextField } from "@fluentui/react";
-import { getDarkGreyTheme, getPrimaryGreenTheme, getPrimaryRedTheme } from "../../../../common/themes";
+import { getDarkGreyTheme, getPrimaryGreenTheme, getPrimaryRedTheme, getPrimaryGreyTheme } from "../../../../common/themes";
```
Comment (alex-krasn) [inline src/react/components/pages/modelCompose/composeModelView.tsx:6]: Screen Shot 2020-07-17 at 3 15 50 PM
****

**Pull request merged**